general/g.region/main.c

```diff
@@ -388,35 +391,42 @@ int main(int argc, char *argv[])
                                   "json;JSON (JavaScript Object Notation);");
     parm.format->guisection = _("Print");
 
-    G_option_required(flag.dflt, flag.savedefault, flag.print, flag.lprint,
-                      flag.eprint, flag.center, flag.gmt_style, flag.wms_style,
-                      flag.dist_res, flag.nangle, flag.z, flag.bbox,
-                      flag.gprint, flag.res_set, flag.noupdate, parm.region,
-                      parm.raster, parm.raster3d, parm.vect, parm.north,
-                      parm.south, parm.east, parm.west, parm.top, parm.bottom,
-                      parm.rows, parm.cols, parm.res, parm.res3, parm.nsres,
-                      parm.ewres, parm.nsres3, parm.ewres3, parm.tbres,
-                      parm.zoom, parm.align, parm.save, parm.grow, NULL);
+    G_option_required(
+        flag.dflt, flag.savedefault, flag.print, flag.lprint, flag.eprint,
+        flag.center, flag.gmt_style, flag.wms_style, flag.dist_res, flag.nangle,
+        flag.z, flag.bbox, flag.gprint, flag.res_set, flag.noupdate,
+        flag.flprint, parm.region, parm.raster, parm.raster3d, parm.vect,
+        parm.north, parm.south, parm.east, parm.west, parm.top, parm.bottom,
+        parm.rows, parm.cols, parm.res, parm.res3, parm.nsres, parm.ewres,
+        parm.nsres3, parm.ewres3, parm.tbres, parm.zoom, parm.align, parm.save,
+        parm.grow, NULL);
     G_option_exclusive(flag.noupdate, flag.force, NULL);
     G_option_requires(flag.noupdate, flag.savedefault, flag.print, flag.lprint,
                       flag.eprint, flag.center, flag.gmt_style, flag.wms_style,
                       flag.dist_res, flag.nangle, flag.z, flag.bbox,
-                      flag.gprint, parm.save, NULL);
-    G_option_requires(flag.flprint, flag.gprint, NULL);
+                      flag.gprint, flag.flprint, parm.save, NULL);
 
     if (G_parser(argc, argv))
         exit(EXIT_FAILURE);
 
     G_get_default_window(&window);
 
-    flat_flag = flag.flprint->answer;
+    if (flag.flprint->answer) {
```

Comment:
If you use format=json and -f together, it will show json, so it doesn't ignore format option. In this case it should just show the flat representation.